ledger-service/http-json/src/main/scala/com/digitalasset/http/EndpointsCompanion.scala

```diff
@@ -334,33 +264,42 @@ object EndpointsCompanion {
 
   private[http] def format(a: JsValue): ByteString = ByteString(a.compactPrint)
 
+  private[http] def decodeAndParseJwt(
+      jwt: Jwt,
+      decodeJwt: ValidateJwt,
+  ): Error \/ AuthServiceJWTPayload =
+    decodeJwt(jwt)
+      .flatMap(a =>
+        AuthServiceJWTCodec
+          .readFromString(a.payload)
+          .toEither
+          .disjunction
+          .leftMap(Error.fromThrowable)
+      )
+
   private[http] def decodeAndParsePayload[A](
       jwt: Jwt,
       decodeJwt: ValidateJwt,
       userManagementClient: UserManagementClient,
       ledgerIdentityClient: LedgerIdentityClient,
   )(implicit
-      lc: LoggingContextOf[InstanceUUID with RequestID],
-      customParse: ParsePayload[A],
+      createFromCustomToken: CreateFromCustomToken[A],
       createFromUserToken: CreateFromUserToken[A],
       fm: Monad[Future],
       ec: ExecutionContext,
-  ): ET[(Jwt, A)] = {
+  ): EitherT[Future, Error, (Jwt, A)] = {
```

Comment:
Any `Error` can occur?

Reply:
Maybe I should have asked the question why the error has to be wider than some authorization error, but it is a minor issue.